internal/transport/grpchttp2/http2bridge_test.go

```diff
@@ -0,0 +1,600 @@
+/*
+ *
+ * Copyright 2024 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package grpchttp2
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"golang.org/x/net/http2/hpack"
+)
+
+// testConn is a test utility which provides an io.Writer and io.Reader
+// interface and access to its internal buffers for testing.
+type testConn struct {
+	wbuf []byte
+	rbuf []byte
+}
+
+func (c *testConn) Write(p []byte) (int, error) {
+	c.wbuf = append(c.wbuf, p...)
+	return len(p), nil
+}
+
+func (c *testConn) Read(p []byte) (int, error) {
+	n := copy(p, c.rbuf)
+	c.rbuf = c.rbuf[n:]
+	return n, nil
+}
+
+func appendUint32(b []byte, x uint32) []byte {
+	return append(b, byte(x>>24), byte(x>>16), byte(x>>8), byte(x))
+}
+
+func readUint32(b []byte) uint32 {
+	return uint32(b[0])<<24 | uint32(b[1])<<16 | uint32(b[2])<<8 | uint32(b[3])
+}
+
+func readUint24(b []byte) int {
+	return int(b[0])<<16 | int(b[1])<<8 | int(b[2])
+}
+
+func readUint16(b []byte) uint16 {
+	return uint16(b[0])<<8 | uint16(b[1])
+}
+
+type wantHeader struct {
+	wantSize      int
+	wantFrameType FrameType
+	wantFlag      Flag
+	wantStreamID  uint32
+}
```

Comment:
Hmm ... This has the exact same fields as that of the exported `FrameHeader` type. I would prefer that we use that instead. So, your helper would be:
```golang
func checkWrittenHeader(gotHdr []byte, wantHdr FrameHeader) []error { ... }
```

Also, places where you access the fields would be `wantHdr.Size` instead of `wantHdr.wantSize`, the latter being more repetitive.

Reply:
I totally missed this overlap. The tests look way cleaner this way. Thanks!